velox/functions/prestosql/tests/ReduceTest.cpp

```diff
@@ -242,3 +243,29 @@ TEST_F(ReduceTest, nullArray) {
   assertEqualVectors(
       makeNullableFlatVector<int64_t>({std::nullopt, std::nullopt}), result);
 }
+
+// Verify limit on the number of array elements.
+TEST_F(ReduceTest, limit) {
+  // Make array vector with huge arrays in rows 2 and 4.
+  auto data = makeRowVector({makeArrayVector(
+      {0, 1'000, 10'000, 100'000, 100'010}, makeConstant(123, 1'000'000))});
+
+  VELOX_ASSERT_THROW(
+      evaluate("reduce(c0, 0, (s, x) -> s + x, s -> s)", data),
```

Comment:
nit: Can we make it something more complex (e.g. `reduce(c0, 0, (s, x) -> s / 2 + x, s -> s)`)?  This could be rewritten to `array_sum` in the future and no longer apply the limit

Reply:
@Yuhta I could make this change, but it makes the code hard to read and understand because it is not obvious what is the expected result. All arrays get reduced to a "magic" value of 245. A reader will need to spend quite some time wondering why that is the case.